saged/models.py

```diff
@@ -467,6 +467,58 @@ def set_final_layer(self, new_layer: nn.Module):
         self.dec = new_layer
 
 
+class FCBlock(nn.Module):
+    def __init__(self, input_size: int, output_size: int):
+        super(FCBlock, self).__init__()
+        DROPOUT_PROB = .5
+        self.fc = nn.Linear(input_size, output_size)
+        self.bn = nn.BatchNorm1d(output_size)
+        self.dropout = nn.Dropout(p=DROPOUT_PROB)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = F.relu(self.fc(x))
+        x = self.bn(x)
+        x = self.dropout(x)
+        return x
+
+
+class GeneralClassifier(nn.Module):
+    def __init__(self,
+                 input_size: int,
+                 intermediate_layers: int,
+                 output_size: int,
+                 **kwargs):
+        """
+        Model initialization function
+
+        Arguments
+        ---------
+        input_size: The number of features in the dataset
+        intermediate_layers: The number of layers in the middle of the model. The total number
+                             of layers will be intermediate_layers + 2
+        output_size: The number of classes to predict
+        """
+        super(GeneralClassifier, self).__init__()
+
+        self.l1 = FCBlock(input_size, input_size // 2)
+
+        intermediate_list = []
+        for _ in range(intermediate_layers):
+            intermediate_list.append(FCBlock(input_size // 2, input_size // 2))
+
+        self.intermediate = nn.ModuleList(intermediate_list)
+
+        self.output = nn.Linear(input_size // 2, output_size)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.l1(x)
+        for layer in self.intermediate:
+            x = layer(x)
+        x = self.output(x)
+
+        return x
```

Comment:
Does the `DeepClassifier` class (below) still have a use? It looks like it's essentially the same as this more general class, but with the number of layers hard-coded instead of passed in as a parameter.

Reply:
Yeah, the five layer network uses the DeepClassifier class as a base

Reply:
Got it! Might be good to have them use the same base class in the future (that way if you make changes to one you won't have to remember to also make changes to the other), but I'm fine with it for now.